zeppelin-zengine/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterProcess.java

```diff
@@ -51,13 +54,27 @@ public void setRemoteInterpreterEventPoller(RemoteInterpreterEventPoller eventPo
     this.remoteInterpreterEventPoller = eventPoller;
   }
 
+  public void shutdown()
+  {
+
+    // Close client socket connection
+    if (clientFactory != null)
+    {
+      for (TSocket eachTransfer: clientFactory.clientSocketMap.values())
```

Comment:
It's better to add `close` method in ClientFactory, can call it here like this
```
if (clientFactory != null) {
   clientFactory.close();
}
```

Reply:
ok